src/app/templates/activity/Activity.tsx

```diff
@@ -1,188 +1,67 @@
+interface Props {
+  assetSlug?: string;
+}
 
-  /**
-   * New operations syncing
-   */
+export const ActivityComponent: React.FC<Props> = ({ assetSlug }) => {
+  const { loading, reachedTheEnd, list: activities, loadMore } = useActivities(INITIAL_NUMBER, assetSlug);
 
-  const syncNewOperations = useCallback(async () => {
-    setSyncing(true);
-    try {
-      const newCount = await syncOperations('new', chainId, address);
-      if (newCount > 0) {
-        refetchLatest();
-      }
-    } catch (err: any) {
-      console.error(err);
-      setSyncError(err);
-    }
-    setSyncing(false);
-  }, [setSyncing, setSyncError, chainId, address, refetchLatest]);
+  const { publicKeyHash: accountAddress } = useAccount();
 
-  const timeoutRef = useRef<any>();
+  if (activities.length === 0 && !loading && reachedTheEnd)
+    return (
+      <div className={classNames('mt-4 mb-12', 'flex flex-col items-center justify-center', 'text-gray-500')}>
+        <LayersIcon className="w-16 h-auto mb-2 stroke-current" />
 
-  const syncAndDefer = useCallback(async () => {
-    await syncNewOperations();
-    timeoutRef.current = setTimeout(syncAndDefer, 10_000);
-  }, [syncNewOperations]);
+        <h3 className="text-sm font-light text-center" style={{ maxWidth: '20rem' }}>
+          <T id="noOperationsFound" />
+        </h3>
+      </div>
+    );
 
-  useEffect(() => {
-    if (syncSupported) {
-      syncAndDefer();
-    }
+  const retryInitialLoad = () => loadMore(INITIAL_NUMBER);
+  const loadMoreActivities = () => loadMore(LOAD_STEP);
 
-    return () => clearTimeout(timeoutRef.current);
-  }, [syncSupported, syncAndDefer]);
+  const loadNext = activities.length === 0 ? retryInitialLoad : loadMoreActivities;
 
   return (
-    <ActivityView
-      address={address}
-      syncSupported={syncSupported}
-      operations={operations ?? []}
-      initialLoading={fetching || (!operations || operations.length === 0 ? syncing : false)}
-      loadingMore={loadingMore}
-      syncing={syncing}
-      loadMoreDisplayed={hasMoreRef.current}
-      loadMore={handleLoadMoreInner}
-      className={className}
-    />
+    <div className="w-full max-w-md mx-auto flex flex-col">
+      <InfiniteScroll
+        dataLength={activities.length}
+        hasMore={reachedTheEnd === false}
+        next={loadNext}
+        loader={loading && <ActivitySpinner height="2.5rem" />}
+        onScroll={({ target }) => {
+          /*
+            In case we catch an error on loading more items, `InfiniteScroll.next` won't trigger.
+            Thus we also force its call here on condition of error.
+          */
+          if (loading || reachedTheEnd) return;
+          const elem: HTMLElement =
+            target instanceof Document ? (target.scrollingElement! as HTMLElement) : (target as HTMLElement);
+          const atBottom = 0 === elem.offsetHeight - elem.clientHeight - elem.scrollTop;
+          if (atBottom) loadNext();
+        }}
+      >
```

Comment:
this function is too complicated to be in component jsx. 
 maybe extract this into `handleScroll` const?